src/main/java/org/mockito/Mockito.java

```diff
@@ -1932,6 +1932,7 @@ public class Mockito extends ArgumentMatchers {
      * @return mock object
      * @since 4.9.0
      */
+    @SafeVarargs
     public static <T> T mock(T... reified) {
```

Comment:
In a sense there [is already a test](https://github.com/mockito/mockito/blob/a9595f559cf4acdaed92371537ced982e30f508d/src/test/java/org/mockito/MockitoTest.java#L164) for this case, but as Mockito currently doesn't compile with something like `options.compilerArgs << "-Xlint:unchecked" << "-Werror"`, the build currently doesn't fail on it. The changes in this PR are relevant for users that do compile with those flags.